pkg/smb/nodeserver_test.go

```diff
@@ -198,108 +200,152 @@ func TestNodeExpandVolume(t *testing.T) {
 }
 
 func TestNodePublishVolume(t *testing.T) {
-	skipIfTestingOnWindows(t)
 	volumeCap := csi.VolumeCapability_AccessMode{Mode: csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER}
-	errorMountSource := "./error_mount_source"
-	alreadyMountedTarget := "./false_is_likely_exist_target"
-	smbFile := "./smb.go"
-	sourceTest := "./source_test"
-	targetTest := "./target_test"
 
+	errorMountSource := map[string]string{
+		"linux":   "./error_mount_source",
+		"windows": "C:\\var\\lib\\kubelet\\error_mount_source",
+	}
+	alreadyMountedTarget := map[string]string{
+		"linux":   "./false_is_likely_exist_target",
+		"windows": "C:\\var\\lib\\kubelet\\fake_is_likely_exist_target",
+	}
+	smbFile := map[string]string{
+		"linux":   "./smb.go",
+		"windows": "C:\\var\\lib\\kubelet\\smb.go",
```

Comment:
why using this path `C:\\var\\lib\\kubelet\\` on windows?

Reply:
another way is using GetWorkDirPath: https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/6e64aea79994672fcc356588d1c652db26fb9136/pkg/azurefile/nodeserver_test.go#L87

Reply:
the reason why I am using that path is because, within the CSI-Proxy logs I get this error: 
```
PS C:\Users\manoh\github\csi-proxy\bin> .\csi-proxy.exe -v 9
I0925 19:31:16.912285   17948 main.go:51] Starting CSI-Proxy Server ...
I0925 19:31:16.913292   17948 main.go:52] Version: v0.2.1-2-g25491d0
I0925 19:31:39.611493   17948 server.go:122] calling PathExists with path "C:\\Users\\manoh\\github\\csi-driver-smb\\pkg\\smb\\smg.go"
E0925 19:31:39.611493   17948 server.go:125] failed validatePathWindows path: C:\Users\manoh\github\csi-driver-smb\pkg\smb\smg.go is not within context path: C:\var\lib\kubelet

```

It explicitly tells here the path is supposed to be `C:\\var\\lib\\kubelet\\`,

Reply:
then what about use func like this, that would be more clear in logic:
```
func getTestPath(path string) string {
	if runtime.GOOS == "windows" {
		return filepath.Join("C:\\var\\lib\\kubelet\\", path)
	}
	return filepath.Join("./", path)
}
```